Example/Controllers/MAFirstLaunchController.swift

```diff
@@ -39,6 +58,80 @@ class MAFirstLaunchController: UIViewController {
+        return permissionsCollections?.count ?? 0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = tableView.dequeueReusableCell(
+            withIdentifier: "FirstLaunchCustomPermissionCell", for: indexPath) as? FirstLaunchCustomPermissionCell {
+                let permissionModel: MASDKCustomPermissionModel?
+                if miniAppManifest?.requiredPermissions?.indices.contains(indexPath.row) ?? false {
+                    permissionModel =  miniAppManifest?.requiredPermissions?[indexPath.row]
+                    cell.permissionTitle?.attributedText =  NSMutableAttributedString()
+                        .normalText(permissionModel?.permissionName.title ?? "")
+                        .highlightRedColor(" (required)")
+                    cell.permissionDescription?.text = permissionModel?.permissionDescription
+                    cell.toggle.isHidden = true
+                } else {
+                    if miniAppManifest?.optionalPermissions?.indices.contains(indexPath.row - (miniAppManifest?.requiredPermissions?.count ?? 0)) ?? false {
+                        permissionModel =  miniAppManifest?.optionalPermissions?[indexPath.row - (miniAppManifest?.requiredPermissions?.count ?? 0)]
+                        cell.permissionTitle?.text = permissionModel?.permissionName.title
+                        cell.permissionDescription?.text = permissionModel?.permissionDescription
+                    }
+                }
+            cell.toggle.tag = indexPath.row
+            cell.toggle.isOn = true
+            cell.toggle.addTarget(self, action: #selector(permissionValueChanged(_:)), for: .valueChanged)
+            return cell
+        }
+        return UITableViewCell()
+    }
+
+    @objc func permissionValueChanged(_ sender: UISwitch) {
+        if permissionsCollections?.indices.contains(sender.tag) ?? false {
+            let permissionModel = permissionsCollections?[sender.tag]
+            if sender.isOn {
+                permissionModel?.isPermissionGranted = .allowed
+            } else {
+                permissionModel?.isPermissionGranted = .denied
+            }
+        }
+    }
+}
+
 protocol MALaunchScreenDelegate: class {
     func didUserResponded(agreed: Bool, miniAppInfo: MiniAppInfo?)
 }
+
+class FirstLaunchCustomPermissionCell: UITableViewCell {
+
+    @IBOutlet weak var permissionTitle: UILabel!
+    @IBOutlet weak var permissionDescription: UILabel!
+    @IBOutlet weak var toggle: UISwitch!
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        toggle.isOn = true
+        permissionTitle.text = ""
+        permissionDescription.text = ""
+    }
+}
+
+extension NSMutableAttributedString {
+    func highlightRedColor(_ value: String) -> NSMutableAttributedString {
+        let attributes: [NSAttributedString.Key: Any] = [
+            .foregroundColor: UIColor.red
```

Comment:
if you wanted to use the crimson red you can access in `self.window?.tintColor` as it is defined in the app delegate

Reply:
Done